### PR TITLE
Chunk saving

### PR DESCRIPTION
**Pull request by koskja**
# Chunk saving

## Status

- [x] Ready 
- [ ] Development
- [ ] Hold

## Description

Chunks now get saved when they are unloaded.
Removed the WorldSource trait and a WorldGenerator and a RegionWorker are now used instead.

## Related issues

#443 

## Checklist

- [ ] Ran `cargo fmt`, `cargo clippy`, `cargo build --release` and `cargo test` and fixed any generated errors!
- [ ] Removed unnecessary commented out code
- [ ] Used specific traces (if you trace actions please specify the cause i.e. the player)

### CODE REVIEW

**Comment by koskja** [comment]
Currently experiencing a bug that causes a chunk to get shifted 16 blocks up when unloaded and loaded later. Seems to involve https://github.com/feather-rs/feather/commit/fcb811de1966f33831413de32951f3550ba3029a

**Comment by koskja** [comment]
Review?

**Review by Defman — COMMENTED** [review]
Looks overall good. The chunk cache `unload_queue` does not seem to getting filled?

Its quite a few changes, but there's nothing jumping out as being plain wrong.

**Comment by koskja** [comment]
Maybe review? Btw, should I just ping some dev every time I make changes?

**Review by caelunshun — APPROVED** [review]
Thank you! Sorry about the delay.

**Comment by caelunshun** [comment]
Looks like a Clippy error is blocking CI.

File: feather/server/src/main.rs [hunk 1]
```diff
@@ -35,17 +29,17 @@ async fn main() -> anyhow::Result<()> {
     let options = config.to_options();
     let server = Server::bind(options).await?;
 
-    let game = init_game(server, config.world)?;
+    let game = init_game(server, &config)?;
```
Comment (poly000) [inline feather/server/src/main.rs:32]:
```rust
let game = init_game(server, &config)?;
```
this borrow operator should be removed

**Review by poly000 — COMMENTED** [review]
fix clippy warning

**Comment by koskja** [comment]
Build passes now

**Pull request merged**